pkg/controller/cyclenoderequest/controller.go

```diff
@@ -85,6 +85,30 @@ func NewReconciler(
 	return reconciler, nil
 }
 
+// Validates the tls configuration for a pre-termination check or healthcheck and
+// returns an error if these are misconfigured
+func tlsCertsValid(tlsConfig v1.TLSConfig) error {
+	// Check that either both the the tls cert and private key are added or missing
+	// If they are both added, cyclops will forward them when making requests for mTLS
+	// If they are both missing, they will not be added. No mTLS.
+	rootCA := os.Getenv(tlsConfig.RootCA)
+	certificate := os.Getenv(tlsConfig.Certificate)
+	key := os.Getenv(tlsConfig.Key)
+
+	if (certificate == "" && key != "") || (certificate != "" && key == "") {
```

Comment:
feels like this can be simplified. essentially it just makes sure that all fields are present?, but with good error messages.

```go
switch {
case certificate == "":
  return fmt.Errorf("certificate is missing")
case key == "":
  return fmt.Errorf("key is missing")
case rootCA == "":
  return fmt.Errorf("rootCA is missing")
default:
  return nil
} 
```

Reply:
So not quite. The root ca can be present without the cert and key. This would be standard tls. The cert and key can optionally be added together which would the host on the node to authenticate Cyclops via mutual TLS.

Reply:
ahh I see. Essentially there's 3 valid "modes" then?
 
1. No Fields - No TLS
2. CA Only - TLS
3. All Fields - mTLS

Reply:
Yes exactly. I'll add this as part of the comment to make it clearer.